File: files/2025-10-05/CSEC 05102025.py
import pandas as pd

path = "files/2025-10-05/Challenge 65.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=1, nrows=13)
test = pd.read_excel(path, usecols="D", skiprows=1, nrows=13)

input['cid'] = (input['Cost Codes'] != input['Cost Codes'].shift()).cumsum()
result = input.assign(
    Consecutive = input.groupby('cid')['Cost Codes'].transform('size') > 1
)[['Consecutive']]

print(result.equals(test)) # True

File: files/2025-10-05/CSEC 05102025.R
library(tidyverse)
library(readxl)

path = "files/2025-10-05/Challenge 65.xlsx"
input = read_excel(path, range = "B2:B14")
test  = read_excel(path, range = "D2:D14")

result = input %>%
  mutate(cid = consecutive_id(`Cost Codes`)) %>%
  mutate(gr_n = n(), .by = cid) %>%
  mutate(Consecutive = ifelse(gr_n > 1, T, F)) %>%
  select(Consecutive)

all.equal(result, test)
# [1] TRUE